app/controllers/bookingsync_portal/admin/rentals_controller.rb

```diff
@@ -43,89 +40,112 @@ def action_variables
 
       public :current_account
 
-      private
-
-      def search_by_rentals?
-        params[:rentals_search].present?
+      def search_by_core_listings?
+        params[:core_listings_search].present?
       end
 
-      def search_by_remote_rentals?
-        params[:remote_rentals_search].present?
+      def search_by_channel_listings?
+        params[:channel_listings_search].present?
       end
 
+      private
+
       def resolve_action
-        redirect_to admin_v2_rentals_path if use_paginated_view?
+        if BookingsyncPortal.use_paginated_view.call(current_account)
+          redirect_to admin_v2_rentals_path
+        end
       end
 
       def prepare_index_variables
         @action_variables = OpenStruct.new
 
-        @action_variables.not_connected_rentals = current_account.rentals.visible.ordered.not_connected
-        @action_variables.visible_rentals = current_account.rentals.visible
-        @action_variables.remote_rentals = current_account.remote_rentals.ordered
-        @action_variables.blank_remote_accounts = generate_blank_remote_accounts
+        @listings_repository = BookingsyncPortal.listings_repository_proc.call(current_account)
+
+        set_core_listings { @listings_repository.find_core_listings }
+        set_available_listings_count { @listings_repository.available_listings_count }
+
+        set_channel_listings { @listings_repository.find_channel_listings }
```

Comment:
cc @kaizencodes 
All these additional wrappers came when I found one performance-related thing: when we use search bar for the core listings we make additional requests for the listing on the right side and vice versa. All these methods were aimed to fix this behavior. Initially I thought that it is required to have such endpoint because you can share the link but link is not overwritten when you do requests inside iframe. So there is no need to always return the actual picture for both lists: it is important to do that in the first request but for further ones we need to just update variables used for one of the lists.

I'm not sure if the implementation is good enough, I think it will be better to add two more controller actions (or two new controllers) in order to completely remove all these checks. But here comes other apps with their additional functionality which is added via `before_rentals_index_action_filter` and `after_rentals_index_action_filter`. So it may be not so easy and clear.  WDYT?